hw_12/cmd/server/main.go

```
@@ -0,0 +1,213 @@
+	}
+
+	record, err := rs.GetRecord(p.Key)
+	if err != nil {
+		return "", err
+	}
+
+	resp := &cmds.GetCommandResponsePayload{
+		Value: record.Value,
+		Ok:    true,
+	}
+
+	rawResp, err := json.Marshal(resp)
+	if err != nil {
+		return "", fmt.Errorf("error marshalling response: %w", err)
+	}
+
+	return string(rawResp), nil
+}
+
+func execDelete(raw string, rs like_mongo.Service) (string, error) {
+	p := &cmds.DeleteCommandRequestPayload{}
+	err := json.Unmarshal([]byte(raw), p)
+	if err != nil {
+		return "", fmt.Errorf("error unmarshalling payload: %w", err)
+	}
+
+	err = rs.DeleteRecord(p.Key)
+	if err != nil {
+		return "", err
+	}
+	resp := &cmds.DeleteCommandResponsePayload{
+		Ok: true,
+	}
+
+	rawResp, err := json.Marshal(resp)
+	if err != nil {
+		return "", fmt.Errorf("error marshalling response: %w", err)
+	}
+
+	return string(rawResp), nil
+}
+
+func execInvalidCommand(raw string) (string, error) {
+	resp := &cmds.ExitCommandResponsePayload{
+		Msg: raw,
+	}
+	rawResp, err := json.Marshal(resp)
+	if err != nil {
+		return "", fmt.Errorf("error marshalling response: %w", err)
+	}
+
+	return string(rawResp), nil
+}
+
+func execList(rs like_mongo.Service) (string, error) {
+	records, err := rs.ListRecords()
+	if err != nil {
+		return "", err
+	}
+	resp := &cmds.ListCommandResponsePayload{
+		Items: records,
+	}
+	rawResp, err := json.Marshal(resp)
+	if err != nil {
+		return "", fmt.Errorf("error marshalling response: %w", err)
+	}
+
+	return string(rawResp), nil
+}
+
+func handleConnection(conn net.Conn) {
+	defer conn.Close()
+
+	logger.Info("connection accepted" + conn.RemoteAddr().String())
+	scanner := bufio.NewScanner(conn)
+	w := bufio.NewWriter(conn)
+
+	var store = document_store.NewStore()
+
+	gofakeit.Seed(0)
+	name := gofakeit.Word()
+	var rs = like_mongo.CreateService(store, "key", name)
+
+	for scanner.Scan() {
+		msg := scanner.Text()
+
+		req, _ := base64.StdEncoding.DecodeString(msg)
+		elems := strings.Split(string(req), " ")
```

don't ignore errors, also this solution doesn't provide safe results as JSON message is base64 encoded along with commands. If there'd be any " " in JSON (for instance in description field with more than 1 word) it will fail. 
Possible solutions here is to base64 only data, but not cmd. 
Another way is to consider JSON messages with commands and data included